### ISSUE TITLE
Evaluate mobilenet v3 keras model performance on ImageNet

### ISSUE THREAD

**Issue by Kewenjing1020** [tensorflow/tensorflow#48066]
Hi, I've tested the keras mobilenetv3_large_1.0 model accuracy, the model is created by calling the official API and using the official h5 model:
`model = tf.keras.applications.MobileNetV3Large(alpha=1.0, minimalistic=False, weights='imagenet')`

I do the following preprocess on ImageNet:
```
1. cropp the image of size [0.875*original height, 0.875*original width] from the center of original image, 
2. resize the image to 224x224
3. each channel subtract 127.5, multiply 2 (as the .h5 model has a rescaling layer, *1/255)
4. convert BGR to RGB format
```
Finally, I call model.evaluate() to do evaluation. But I can only get the accuracy:
>  sparse_categorical_accuracy: 0.7208 - sparse_top_k_categorical_accuracy: 0.9122

which is 3% less than reported 75%. Could you help me some advice on how the get the right accuracy?

**Comment by Saduf2019** [comment]
@Kewenjing1020 

We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]

**Comment by Kewenjing1020** [comment]
Here's my validation code:
```
import os
import tensorflow as tf
import tensorflow.keras as keras
from tensorflow.keras.optimizers import RMSprop
from ipdb import set_trace
import numpy as np
import cv2

model = tf.keras.applications.MobileNetV3Large(alpha=1.0, minimalistic=False, weights='/group/modelzoo/internal-cooperation-models/tensorflow/classification_tf_2/mobilenet_v3/float/weights_mobilenet_v3_large_224_1.0_float.h5')
i = tf.keras.layers.Input([None, None, 3], dtype = tf.uint8)
x = tf.cast(i, tf.float32)
x = model(x)
my_model = tf.keras.Model(inputs=[i], outputs=[x])

eval_image_list='/group/modelzoo/test_dataset/Imagenet/val.txt'
eval_image_path='/group/modelzoo/test_dataset/Imagenet/val_dataset'

with open(eval_image_list, 'r') as fr:
    lines = fr.readlines()

#def inception_preprocess(image, central_fraction=0.875,
def inception_preprocess(img_path, central_fraction=0.875,
                                     central_crop=True):
    image = tf.io.read_file(img_path)
    image = tf.image.decode_jpeg(image, channels=3)
    image = tf.cast(image,tf.float32)
    if central_crop and central_fraction:
      image = tf.image.central_crop(image, central_fraction=central_fraction)

    image = tf.expand_dims(image, 0)
    image = tf.compat.v1.image.resize_bilinear(image,[224,224])
    image = tf.subtract(image, 127.5)
    image = tf.multiply(image, 2.0)
    return image


opt = RMSprop(learning_rate=1e-5)
loss = tf.losses.SparseCategoricalCrossentropy()
metric_top_5 = keras.metrics.SparseTopKCategoricalAccuracy()
accuracy = keras.metrics.SparseCategoricalAccuracy()
my_model.compile(optimizer=opt, loss=loss, metrics=[accuracy, metric_top_5])

batch = 1000
top1_accs = []
top5_accs = []
for n in range(len(lines)//batch+1):
    imagelist = []
    labels = []
    for k,line in enumerate(lines[n*batch:min((n+1)*batch,len(lines))]):
        img_path, label = lines[k].strip().split(" ")
        img_path = os.path.join(eval_image_path, img_path)
        label = np.array([label], dtype=np.int64)
        image = inception_preprocess(img_path)
        imagelist.append(image)
        labels.append(label)

    _,top1,top5= my_model.evaluate(keras.layers.Concatenate(0)(imagelist),np.array(labels))
    top1_accs.append(top1*len(labels))
    top5_accs.append(top5*len(labels))

print('top1:',np.array(top1_accs).sum()/len(lines))
print('top5:',np.array(top5_accs).sum()/len(lines))

```
With this code, I can only achieve top1 ~50%. My environment is : tensorflow==2.4.0, python=3.8.2, cuda-version 11.2

**Comment by Kewenjing1020** [comment]
@Saduf2019 The error happens because you don't install ipdb. Just install ipdb by 'pip install ipdb'. Or, remove 'from ipdb import set_trace'.
Besides, in this code, it calls the Imagenet dataset by giving the datapath and vallist.  You need to change the 'eval_image_list' and 'eval_image_path'. The eval_image_list file is defined as:
```
ILSVRC2012_val_00000001.JPEG 65
ILSVRC2012_val_00000002.JPEG 970
ILSVRC2012_val_00000003.JPEG 230
ILSVRC2012_val_00000004.JPEG 809
ILSVRC2012_val_00000005.JPEG 516
ILSVRC2012_val_00000006.JPEG 57
ILSVRC2012_val_00000007.JPEG 334
ILSVRC2012_val_00000008.JPEG 415
ILSVRC2012_val_00000009.JPEG 674
ILSVRC2012_val_00000010.JPEG 332
ILSVRC2012_val_00000011.JPEG 109
ILSVRC2012_val_00000012.JPEG 286
ILSVRC2012_val_00000013.JPEG 370
ILSVRC2012_val_00000014.JPEG 757
ILSVRC2012_val_00000015.JPEG 595
ILSVRC2012_val_00000016.JPEG 147
ILSVRC2012_val_00000017.JPEG 108
ILSVRC2012_val_00000018.JPEG 23
ILSVRC2012_val_00000019.JPEG 478
ILSVRC2012_val_00000020.JPEG 517
ILSVRC2012_val_00000021.JPEG 334
ILSVRC2012_val_00000022.JPEG 173
ILSVRC2012_val_00000023.JPEG 948
ILSVRC2012_val_00000024.JPEG 727
ILSVRC2012_val_00000025.JPEG 23
ILSVRC2012_val_00000026.JPEG 846
ILSVRC2012_val_00000027.JPEG 270
ILSVRC2012_val_00000028.JPEG 167
ILSVRC2012_val_00000029.JPEG 55
ILSVRC2012_val_00000030.JPEG 858
ILSVRC2012_val_00000031.JPEG 324
ILSVRC2012_val_00000032.JPEG 573
ILSVRC2012_val_00000033.JPEG 150
ILSVRC2012_val_00000034.JPEG 981
ILSVRC2012_val_00000035.JPEG 586
```

**Comment by Saduf2019** [comment]
@Kewenjing1020 
Please share all dependencies for us to replicate the issue, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/5c7073a8b1586100ef96a27ebf36a82e/untitled586.ipynb) or if possible share a colab gist with the issue reported.

**Comment by qlzh727** [comment]
Thanks for reporting the issue. I wasn't able to access the data file you are using in the colab.

Having said that, the mobilenet v3 include the preprocessing logic in the model and was expecting the input data to be ranged from [0, 255]. Since I can't access your data file, I can't tell whether you logic in inception_preprocess is correct (it seems to make the data ranged from [-255, 255], assuming the original data is from [0, 255]). Could u update that and test again? 

See https://github.com/tensorflow/tensorflow/blob/5dcfc51118817f27fad5246812d83e5dccdc5f72/tensorflow/python/keras/applications/mobilenet_v3.py#L143

**Comment by qlzh727** [comment]
Thanks for reporting the issue. Please https://github.com/tensorflow/tensorflow/pull/48542#issuecomment-824258134 for more updates. #48542 should fix the issue.
